### DISCUSSION TITLE
[Mongo services] Connection through a certificate instead of credentials

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#545] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

v13

### Description

Hi guys,
a client wants to deprecate <username,password> users on mongo and use the certificate for the authentication.

The certificate is currently supported in the mongo connection string, following an example:
> `mongodb+srv://my-secure-cluster.mongodb.net/?authSource=%24external&authMechanism=MONGODB-X509&&tlsCertificateKeyFile=/path/to/my/certificate.pem`

The client uses the crud-service and the Fast Data services.
Is it possible to authenticate in this way?

I am thinking that, if you inject a secret with the certificate inside the specific pod and than you point to that certificate in the connection string all should work well, but I'd like to have a feedback from you too.

Thanks 😄

**Comment by flower-of-the-bridges** [comment, marked as answer]
Hi @silversoul93 

Both `crud-service` and fast data services accepts a valid MongoDB connection string. So, if you mount that certificate as a secret (using [mlp](https://docs.mia-platform.eu/docs/console/handbooks/traefik-handbook#generate-the-secret-with-mlp)), the connection string without username and password should work

**Reply by frengooo** [reply]
Hi @flower-of-the-bridges, that was exactly what I supposed, thanks 😄
